tests/core/v5_1/alexandria/test_proof_serialization.py

```diff
@@ -0,0 +1,149 @@
+        ((True,), VALUE, None,),
+        ((False,), VALUE, None,),
+        ((True,) * 25, VALUE, None,),
+        ((False,) * 25, VALUE, None,),
+        ((False, False, True, True, False, False), VALUE, None,),
+        ((True, True, False, False, True, True), VALUE, None,),
+        # Previous path without any common bits
+        ((True, False, True, False, True, False), VALUE, (False, True),),
+        # Previous path without one common bit
+        ((True, False, True, False, True, False), VALUE, (True, True),),
+        # Previous path without multiple common bits
+        (
+            (True, False, True, False, True, False),
+            VALUE,
+            (True, False, True, False, False),
+        ),
+        # Exceed single byte boundaries
+        (
+            (
+                True,
+                False,
+                True,
+                False,
+                True,
+                False,
+                True,
+                False,
+                True,
+                False,
+                True,
+                False,
+            ),
+            VALUE,
+            (True, False),
+        ),
+        (
+            (
+                True,
+                False,
+                True,
+                False,
+                True,
+                False,
+                True,
+                False,
+                True,
+                False,
+                True,
+                False,
+            ),
+            VALUE,
+            (True, False, True, False, True, False, True, False, True, False),
+        ),
+    ),
+)
+def test_proof_element_serialization_round_trip(path, value, previous):
+    path = tuple(path)
+    element = ProofElement(path, value)
+    serialized = element.serialize(previous)
+    result = ProofElement.deserialize(io.BytesIO(serialized), previous)
+    assert result == element
+
+
+@given(data=st.data(),)
+def test_proof_element_serialization_round_trip_fuzzy(data):
+    path = data.draw(st.lists(st.booleans(), min_size=0, max_size=25).map(tuple))
+    value = data.draw(st.binary(min_size=32, max_size=32))
+    previous = data.draw(
+        st.one_of(
+            st.lists(st.booleans(), min_size=0, max_size=25 - len(path)).map(tuple),
+            st.none(),
+        )
+    )
+
+    element = ProofElement(path, value)
+    serialized = element.serialize(previous)
+    result = ProofElement.deserialize(io.BytesIO(serialized), previous)
+    assert result == element
+
+
+@settings(max_examples=1000)
+@given(content=st.binary(min_size=0, max_size=GB))
+def test_full_proof_serialization_and_deserialization(content):
+    proof = compute_proof(content, sedes=content_sedes)
+
+    serialized = proof.serialize()
+    result = Proof.deserialize(io.BytesIO(serialized), proof.hash_tree_root)
+
+    validate_proof(result)
+
+    assert result == proof
+
+
+@settings(max_examples=1000)
+@given(data=st.data())
+def test_partial_proof_serialization_and_deserialization(data):
```

Comment:
🚀 this test helped to grok all. 

I know the roundtrip/integration tests are probably good enough at this stage in the game, but eventually I think the unit testing of the `serialize` and `deserialize` funcs will help with onboarding.